support/k8s/k8s-workload-registrar/mode-crd/config/spire-server-registrar.yaml

```diff
@@ -0,0 +1,341 @@
+  verbs: ["get", "list", "watch"]
+- apiGroups: ["spiffeid.spiffe.io"]
+  resources: ["spiffeids"]
+  verbs: ["create", "delete", "get", "list", "patch", "update", "watch"]
+- apiGroups: ["spiffeid.spiffe.io"]
+  resources: ["spiffeids/status"]
+  verbs: ["get", "patch", "update"]
+
+---
+
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: k8s-workload-registrar-role-binding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: k8s-workload-registrar-role
+subjects:
+- kind: ServiceAccount
+  name: spire-server
+  namespace: spire
+
+---
+
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: k8s-workload-registrar
+  namespace: spire
+data:
+  k8s-workload-registrar.conf: |
+    log_level = "debug"
+    trust_domain = "example.org"
+    agent_socket_path = "/run/spire/sockets/agent.sock"
+    server_socket_path = "/tmp/spire-registration.sock"
+    cluster = "demo-cluster"
+    pod_controller = true
+    add_svc_dns_names = true
+    mode = "crd"
+    webhook_enabled = true
+    pod_annotation = "spiffe.io/spiffe-id"
+
+---
+
+apiVersion: admissionregistration.k8s.io/v1beta1
+kind: ValidatingWebhookConfiguration
+metadata:
+  name: k8s-workload-registrar
+  labels:
+    spiffe.io/webhook: "true"
+webhooks:
+- name: k8s-workload-registrar.spire.svc
+  clientConfig:
+    service:
+      name: k8s-workload-registrar
+      namespace: spire
+      path: "/validate-spiffeid-spiffe-io-v1beta1-spiffeid"
+  rules:
+  - apiGroups: ["spiffeid.spiffe.io"]
+    apiVersions: ["v1beta1"]
+    operations: ["CREATE", "UPDATE", "DELETE"]
+    resources: ["spiffeids"]
+    scope: Namespaced
+
+---
+
+apiVersion: apps/v1
+kind: StatefulSet
+metadata:
+  name: spire-server
+  namespace: spire
+  labels:
+    app: spire-server
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: spire-server
+  serviceName: spire-server
+  template:
+    metadata:
+      namespace: spire
+      labels:
+        app: spire-server
+    spec:
+      serviceAccountName: spire-server
+      shareProcessNamespace: true
+      containers:
+        - name: spire-server
+          image: gcr.io/spiffe-io/spire-server:0.12.2
+          args:
+            - -config
+            - /run/spire/config/server.conf
+          livenessProbe:
+            exec:
+              command: ["/opt/spire/bin/spire-server", "healthcheck"]
+            failureThreshold: 2
+            initialDelaySeconds: 15
+            periodSeconds: 60
+            timeoutSeconds: 3
+          readinessProbe:
+            exec:
+              command: ["/opt/spire/bin/spire-server", "healthcheck", "--shallow"]
+            initialDelaySeconds: 5
+            periodSeconds: 5
+          ports:
+            - containerPort: 8081
+          lifecycle:
+            preStop:
+              exec:
+                command:
+                   - "/bin/sh"
+                   - "-c"
+                   - "PID=$(pidof k8s-workload-registrar) ; wait $PID"
```

Comment:
The prestop hook is used to ensure the entry created for the webhook is cleaned up. It tell the spire server to wait for the k8s-workload-registrar to exit first.